sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -625,6 +625,7 @@ def _generic_delete_blob_options(delete_snapshots=False, **kwargs):
             delete_snapshots = DeleteSnapshotsOptionType(delete_snapshots)
         options = {
             'timeout': kwargs.pop('timeout', None),
+            'snapshot': kwargs.pop('snapshot', None),  # this is added for delete_blobs
```

Comment:
Is this only used during batch delete?

Reply:
yes, for normal delete. snapshot will be overwritten by self.snapshot